app/assets/javascripts/lara-api.js

```diff
@@ -149,5 +149,119 @@ window.LARA = {
   ****************************************************************************/
   saveUserState: function (pluginInstance, state) {
     console.log('Plugin', pluginInstance, 'wants to save a state:', state);
+  },
+
+
+  /****************************************************************************
+  @function decorateContent: Ask LARA to decorate authored content (text / html)
+  @arg {string[]} words - a list of case-insensitive words to be decorated
+      can use limited regex
+  @arg {string} replace - the replacement string.
+      Can include '$1' representing the matched word.
+  @arg {IEventListeners} listeners - one or more { type, listener } tuples
+  interface IEventListener {
+    type: string;
+    listener: (evt: Event) => void;
+  }
+
+  type IEventListeners = IEventListener | IEventListener[];
+  @returns void
+  ****************************************************************************/
+  decorateContent: function (words, replace, wordClass, listeners) {
+    var domClasses = ['question-txt', 'intro-txt'];
+    var options = {
+      words: words,
+      replace: replace
+    }
+    TextDecorator.decorateDOMClasses(domClasses, options, wordClass, listeners);
+    var debuggingListener = {
+      type: 'click',
+      listener: function (evt)  {
+        alert("You clicked: ", evt.target.textContent, "!");
+      }
+    };
+    TextDecorator.addEventListeners(wordClass, [debuggingListener]);
+  },
+
+
+
+
+  /****************************************************************************
+   private variables to keep track of our plugins.
+  ****************************************************************************/
+  _pluginClasses: {},
+  _plugins:[],
+  _plugin_labels:[],
+
+  _pluginError: function(e, other) {
+    console.group("LARA Plugin Error");
+    console.error(e);
+    console.dir(other);
+    console.groupEnd();
+  },
+
+  _nameForPlugin: function(plugin) {
+    return(plugin.name || "(unknown)");
+  },
+
+  /**************************************************************
+  @function init
+  This method is called to initialize the external scripts
+  Called at runtime by LARA to create an instance of the plugin
+  as would happen in `views/plugin/_show.html.haml`
+  @arg {string} label - the the script identifier.
+  @arg {IRuntimContext} runtimeContext - context for the plugin
+
+  // Its likely we dont need most of these context values. TBD.
+  Interface IRuntimeContext {
+    name: string;         // Name of the plugin
+    url: string;          // Url from which the plugin was loaded
+    pluginId: string;     // Active record ID of the plugin scope id
+    authorData: string;   // The authored configuration for this instance
+    learnerData: string;  // The saved learner data for this instance
+    div: DomNode;         // reserved DomNode for the plugin in ther runtime.
```

Comment:
I guess that's HTMLElement in TypeScript. :) No need to fix it now, but just for the future reference. I'm assuming that one beautiful day we can really copy-paste these interfaces and use them.